Modules/CTP/src/RawDataQcTask.cxx

```diff
@@ -93,40 +97,42 @@ void CTPRawDataReaderTask::startOfActivity(const Activity& activity)
   }
   /// the ccdb reading to be futher discussed
   // o2::ctp::CTPRunManager::setCCDBHost(ccdbName);
-  bool ok;
-  // o2::ctp::CTPConfiguration CTPconfig = o2::ctp::CTPRunManager::getConfigFromCCDB(mTimestamp, run, ok);
-  auto& mgr = o2::ccdb::BasicCCDBManager::instance();
-  mgr.setURL(ccdbName);
-  map<string, string> metadata; // can be empty
-  metadata["runNumber"] = run;
-  auto ctpconfigdb = mgr.getSpecific<o2::ctp::CTPConfiguration>(o2::ctp::CCDBPathCTPConfig, mTimestamp, metadata);
-  if (ctpconfigdb == nullptr) {
-    LOG(info) << "CTP config not in database, timestamp:" << mTimestamp;
-    ok = 0;
-  } else {
-    // ctpconfigdb->printStream(std::cout);
-    LOG(info) << "CTP config found. Run:" << run;
-    ok = 1;
-  }
-  if (ok) {
-    // get the index of the MB reference class
-    ILOG(Info, Support) << "CTP config found, run:" << run << ENDM;
-    std::vector<o2::ctp::CTPClass> ctpcls = ctpconfigdb->getCTPClasses();
-    for (size_t i = 0; i < ctpcls.size(); i++) {
-      classNames[i] = ctpcls[i].name.c_str();
-      if (ctpcls[i].name.find(MBclassName) != std::string::npos) {
-        mIndexMBclass = ctpcls[i].getIndex() + 1;
-        break;
+  if (!mReadCTPconfigInMonitorData) {
+    bool ok;
+    auto& mgr = o2::ccdb::BasicCCDBManager::instance();
+    mgr.setURL(ccdbName);
+    map<string, string> metadata; // can be empty
+    metadata["runNumber"] = run;
+    auto ctpconfigdb = mgr.getSpecific<o2::ctp::CTPConfiguration>(o2::ctp::CCDBPathCTPConfig, mTimestamp, metadata);
+    if (ctpconfigdb == nullptr) {
```

Comment:
Is there a reason not to use a DPL input ?

Reply:
For the moment we keep both options - to read directly and to read from dpl.
There is config parameter `readCTPconfigInMonitorData` which control these options.
The dpl part starts in:[ dpl
](https://github.com/lhusova/QualityControl/blob/27140df1a78f1f16775b4577a5b06b959b190422/Modules/CTP/src/RawDataQcTask.cxx#L235)